### DISCUSSION TITLE
Running didjvu in Thonny (or other debugger)

### DISCUSSION THREAD

**Discussion by jsbien** [FriedrichFroebel/didjvu#23] [Q&A] [answered]
I finally started to learn Python :-) I would like to get a better understanding of didjvu by running it in a debugger. Up to now I used only Thonny. I hope to learn also Code in some future. 
My brute force approach to run `home/jsbien/work/gamera4didjvu/gamera-4/venv/bin/didjvu` (renamed to `didjvu.py` to make it visible to Thonny) failed with the message
`ModuleNotFoundError: No module named 'didjvu.__main__'; 'didjvu' is not a package`
I will appreciate very much your advice.

**Comment by FriedrichFroebel** [comment]
IDEs do not use your own virtual environments by default. Thus you probably have to switch the Python interpreter from the system one to the one inside your virtual environment to correctly identify the installed *didjvu* and Gamera packages: https://www.techcoil.com/blog/how-to-associate-a-python-3-virtual-environment-with-thonny/

**Comment by jsbien** [comment, marked as answer]
Thanks for the suggestion. I had the virtual environment associated with Thonny and used it succesfully, but the association somehow vanished. I re-associated the environment following https://github.com/thonny/thonny/issues/3014.
Renaming .../venv/bin/didjvu to didjvu.py to make it visible to Thonny broke the program causing "circular import". I had to open didjvu using View -Files- Open in Tommy. So the problem is solved.
